### PR TITLE
Added Reference section, Glossary, removed non-inclusive language.

### PR DESCRIPTION
**Pull request by dwelsch-esi**
- Added Reference section to table of contents, issue #1366
    - Moved some content from KEDA concept topics to Reference.
- Added a glossary, issue #1367
- Removed non-inclusive language, issue #1373

https://github.com/kedacore/keda-docs/issues/1366
https://github.com/kedacore/keda-docs/issues/1367
https://github.com/kedacore/keda-docs/issues/1373

Umbrella issue for CNCF tech docs recommendations: https://github.com/kedacore/keda-docs/issues/1361

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Hey, mind fixing the build issue?

Thanks for opening the PR!

**Comment by nate-double-u** [comment]
I've taken a pass at fixing the build errors over on https://github.com/dwelsch-esi/keda-docs/pull/1 @dwelsch-esi, i'm not sure that _this_ PR introduced all of the broken links though. Regardless, I hope the PR into your fork fixes them 🙂

**Comment by tomkerkhove** [comment]
Thank you!

**Comment by thisisobate** [comment]
It looks like we are missing to update newly introduced files to the **2.14** directory. Given that **2.14** is the latest version, all new additions to the next version (v2.15) must be backported to 2.14 for CI to pass.

Also, there's no need to update the `params.version` array in `config.toml`; this should only be edited during the release cycle by a release engineer.

Will push some changes to fix CI right away.

**Comment by nate-double-u** [comment]
Thanks @thisisobate. @tomkerkhove would you like to take a look now the build's not broken 😊

**Comment by tomkerkhove** [comment]
I will do so ASAP, but can you elaborate on the changes in 2.14 which only adds things and not change other parts? Maybe it's best if we just focus on 2.15 and move forward

**Comment by thisisobate** [comment]
@tomkerkhove The top level nav contains an FAQ link which reference the latest version. Making changes to just 2.15 will cause CI to break because the url structure is not the same: `docs/2.14/faq` vs `docs/2.15/reference/faq`.

**Comment by thisisobate** [comment]
@tomkerkhove Honestly, I wouldn't think much about this implementation since it only fixes the CI. All the new changes to directory structure and docs will only reflect in 2.14 (latest version) and beyond.

**Comment by tomkerkhove** [comment]
> @tomkerkhove Honestly, I wouldn't think much about this implementation since it only fixes the CI. All the new changes to directory structure and docs will only reflect in 2.14 (latest version) and beyond.

So if I get it correctly, we are only adding the new pieces to make CI happy due to new navigation, is that correct? If so, then why don't we delete the old locations in 2.14? If somebody needs to update docs for 2.14 (ie typo or wrong content) then they need to update 2 places (and know there are 2).

**Comment by thisisobate** [comment]
> So if I get it correctly, we are only adding the new pieces to make CI happy due to new navigation, is that correct?

Yes, that is correct

**Comment by thisisobate** [comment]
> why don't we delete the old locations in 2.14? If somebody needs to update docs for 2.14 (ie typo or wrong content) then they need to update 2 places (and know there are 2)

what locations are you referring to? can you give me examples?

**Comment by tomkerkhove** [comment]
Pretty much all of the content in 2.14, for example:
- You introduced `content/docs/2.14/operate/cloud-events.md`
- This is already in `content/docs/2.14/operate/events.md`

**Comment by thisisobate** [comment]
Oh…I see. You’re correct. We should remove duplicates/files in old location.

cc: @dwelsch-esi <https://github.com/dwelsch-esi>

File: content/docs/2.14/operate/cloud-events.md
Comment (tomkerkhove) [inline content/docs/2.14/operate/cloud-events.md:112]: Are we purposely keeping it here and not reference?

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.14/reference/_index.md:10]: Dead link?

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.14/reference/_index.md:8]: What is the logic of ordering here?

File: content/docs/2.15/_index.md
Comment (tomkerkhove) [inline content/docs/2.15/_index.md:11]: Let's use the existing formatting, likely `#` or `##` depending on the header level you want

File: content/docs/2.15/_index.md
Comment (tomkerkhove) [inline content/docs/2.15/_index.md:20]: Thoughts on things to add @kedacore/keda-maintainers ?

File: content/docs/2.15/concepts/external-scalers.md
Comment (tomkerkhove) [inline content/docs/2.15/concepts/external-scalers.md:47]: Should we use relative links instead?

File: content/docs/2.15/concepts/scaling-deployments.md
Comment (tomkerkhove) [inline content/docs/2.15/concepts/scaling-deployments.md:32]: Given this used to be hear, maybe we need `# Specification` which simply points to the reference as mentioned above?

File: content/docs/2.15/concepts/scaling-jobs.md
Comment (tomkerkhove) [inline content/docs/2.15/concepts/scaling-jobs.md:20]: Same here about spec section

File: content/docs/2.15/deploy.md
Comment (tomkerkhove) [inline content/docs/2.15/deploy.md:9]: Shouldn't we link to the items?

File: content/docs/2.15/operate/cloud-events.md
Comment (tomkerkhove) [inline content/docs/2.15/operate/cloud-events.md:112]: Move to reference similar to Kubernetes events?

File: content/docs/2.15/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.15/reference/_index.md:10]: Broken link + let's clarify that this is for Kubernetes events

File: content/docs/2.15/reference/events.md
Comment (tomkerkhove) [inline content/docs/2.15/reference/events.md:2]: This is just for Kubernetes events, CloudEvents is still under operate - What is reasoning here?

File: content/docs/2.15/reference/glossary.md
Comment (tomkerkhove) [inline content/docs/2.15/reference/glossary.md:36]:
```suggestion
## gRPC Remote Procedure Calls (gRPC)
```

File: content/docs/2.15/reference/glossary.md
Comment (tomkerkhove) [inline content/docs/2.15/reference/glossary.md:38]:
```suggestion
gRPC Remote Procedure Calls (gRPC). An open-source remote procedure call framework used by KEDA components to communicate.
```

**Review by tomkerkhove — COMMENTED** [review]
Added a few comments, let me know what you think

File: content/docs/2.15/operate/cloud-events.md
Comment (dwelsch-esi) [reply content/docs/2.15/operate/cloud-events.md:112]: Yes. See my comments in reference/events.md

File: content/docs/2.15/reference/_index.md
Comment (dwelsch-esi) [reply content/docs/2.15/reference/_index.md:10]: Or include all events. Are there other options besides Kubernetes and cloud? Or the list too open-ended to keep a reference up to date?

**Comment by tomkerkhove** [comment]
@dwelsch-esi Are you still working on this or waiting for a re-review? I am asking since some things are still open and nothing was pushed.

File: content/docs/2.15/reference/events.md
Comment (tomkerkhove) [reply content/docs/2.15/reference/events.md:2]: I think we should have a table for it as well indeed.

However, given scenarios are different would it not be better to split this in to two tables to emphasize that?

File: content/docs/2.15/concepts/scaling-jobs.md
Comment (tomkerkhove) [reply content/docs/2.15/concepts/scaling-jobs.md:20]: I meant same as https://github.com/kedacore/keda-docs/pull/1389#discussion_r1630984363, sorry.

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [reply content/docs/2.14/reference/_index.md:8]: Agreed

File: content/docs/2.15/concepts/scaling-deployments.md
Comment (dwelsch-esi) [reply content/docs/2.15/concepts/scaling-deployments.md:32]: @tomkerkhove We reference the spec at the top of this page. Maybe the thing to do is to make it more prominent there.

Also, do the same on the Scaling Jobs page.

File: content/docs/2.15/concepts/scaling-jobs.md
Comment (dwelsch-esi) [reply content/docs/2.15/concepts/scaling-jobs.md:20]: Got it. See comment [there](https://github.com/kedacore/keda-docs/pull/1389#discussion_r1630984363).

File: content/docs/2.15/reference/_index.md
Comment (dwelsch-esi) [inline content/docs/2.15/reference/_index.md:10]:
```suggestion
- [ScaledJob specification](./scaledjob-spec)
- [Kubernetes Events](./events)
```

File: content/docs/2.15/reference/events.md
Comment (dwelsch-esi) [reply content/docs/2.15/reference/events.md:2]: If that makes more sense, absolutely. My criterion would be: Does the user normally want the information in one table but not both? Is that what you mean by different scenarios?

File: content/docs/2.15/reference/_index.md
Comment (dwelsch-esi) [reply content/docs/2.15/reference/_index.md:10]: @tomkerkhove @nate-double-u @thisisobate I thought there were still merge issues due to failed checks. Have those been resolved? 

I'm good with the content. I think we've resolved most of the review questions you had. If we merge now, any corrections we have to make will be minor.

File: content/docs/2.14/reference/_index.md
Comment (dwelsch-esi) [reply content/docs/2.14/reference/_index.md:10]:
```suggestion
- [Events](./events)
```

File: content/docs/2.15/concepts/scaling-deployments.md
Comment (dwelsch-esi) [inline content/docs/2.15/concepts/scaling-deployments.md:6]:
```suggestion
This page describes the deployment scaling behavior of KEDA. 

# Specification

See the [Scaled Object specification](../reference/scaledobject-spec.md) for details on how to set the behaviors described below.
```

File: content/docs/2.15/concepts/scaling-jobs.md
Comment (dwelsch-esi) [inline content/docs/2.15/concepts/scaling-jobs.md:6]: This page describes the job scaling behavior of KEDA. 

## Specification

See the [Scaled Job specification](../reference/scaledjob-spec.md) for details on how to set the behaviors described below.

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.14/reference/_index.md:8]: Massive nit
```suggestion
- [ScaledObject specification](./scaledobject-spec)
```

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.14/reference/_index.md:11]: Broken link

File: content/docs/2.14/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.14/reference/_index.md:10]: Shouldn't you add .md at the end?

File: content/docs/2.15/reference/_index.md
Comment (tomkerkhove) [inline content/docs/2.15/reference/_index.md:11]: Same comments here

File: content/docs/2.15/reference/_index.md
Comment (dwelsch-esi) [inline content/docs/2.15/reference/_index.md:8]:
```suggestion
- [ScaledObject specification](./scaledobject-spec)
```

File: content/docs/2.15/reference/_index.md
Comment (dwelsch-esi) [reply content/docs/2.15/reference/_index.md]: Fixed.

File: content/docs/2.14/reference/_index.md
Comment (dwelsch-esi) [reply content/docs/2.14/reference/_index.md]: You would think so, but I investigated this earlier and it seems as if the site generator handles this automatically for relative paths. I keep missing instances (like the above). File extensions are so common that they're almost invisible ...

File: content/docs/2.14/reference/_index.md
Comment (dwelsch-esi) [inline content/docs/2.14/reference/_index.md:9]:
```suggestion
- [ScaledJob specification](./scaledjob-spec)
```

File: content/docs/2.14/reference/_index.md
Comment (dwelsch-esi) [inline content/docs/2.14/reference/_index.md:11]:
```suggestion
- [Firewall requirements](../operate/cluster#firewall)
```

File: content/docs/2.14/reference/_index.md
Comment (nate-double-u) [reply content/docs/2.14/reference/_index.md]: According to the build log, https://app.netlify.com/sites/keda/deploys/667af00917bae9000859d0c6, the link checker seems to not care much between .md and not. IIRC, Hugo prefers .md not be present for internal links.

**Review by nate-double-u — APPROVED** [review]
This looks good to me, the content is ready to go in now, and any further updates that might be needed are small enough to manage in follow up PRs.

**Comment by tomkerkhove** [comment]
Thanks, I agree but wanted to hear from @zroubalik or @JorTurFer first

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Thanks @dwelsch-esi!

**Pull request merged**
